aptos-move/framework/aptos-stdlib/sources/type_map.move

```diff
@@ -0,0 +1,38 @@
+module aptos_std::type_map {
+    use aptos_std::simple_map::{Self, SimpleMap};
+    use aptos_std::type_info::{Self, TypeInfo};
+    use aptos_std::any;
+
+    use std::bcs;
+
+    struct TypeMap has drop, copy, store {
+        inner: SimpleMap<TypeInfo, vector<u8>>
```

Comment:
what if I have multiple instance of the same type? should this be vector<vector<u8>>?

Reply:
The idea is there is one entry per type. This is _exactly_ the same as Move global memory via borrow_global et. al. However, it's not that Move has invented this. The pattern is around since very, very long for expressing heterogenous collections.